api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -214,13 +216,24 @@ func (c *imageBuilder) imageRefExists(imageName, imageTag string) (bool, error)
 	tags, err := remote.List(repo, remote.WithAuthFromKeychain(keychain))
 	if err != nil {
 		if terr, ok := err.(*transport.Error); ok {
+			log.Errorf("transport error on listing tags for %s: status code: %d, error: %s", imageName, terr.StatusCode, terr.Error())
+
 			// If image not found, it's not exist yet
 			if terr.StatusCode == http.StatusNotFound {
 				return false, nil
 			}
-		} else {
-			return false, errors.Wrapf(err, "error getting image tags for %s", repo)
+
+			// If unauthorized, it's more likely due to expired token. So, let's try again
+			if terr.StatusCode == http.StatusUnauthorized {
+				if retryCounter <= maxCheckImageRetry {
```

Comment:
should it be < instead of <=

Reply:
I put `<=` because I want to retry twice. Otherwise it's only retry once.

Reply:
it will retry thrice if you use <= . 
let simulate
 imageRefExist(retry=0)
     failed -> imageRefExist(retry=1)
                     failed -> imageRefExist(retry=2)
                                       failed -> imageRefExist(retry=3)
                                                         failed -> no retry

Reply:
Woop. You're right. Updated as suggested